netty/src/main/java/io/grpc/netty/NettyServer.java

```diff
@@ -144,32 +157,49 @@
     this.permitKeepAliveTimeInNanos = permitKeepAliveTimeInNanos;
     this.eagAttributes = checkNotNull(eagAttributes, "eagAttributes");
     this.channelz = Preconditions.checkNotNull(channelz);
-    this.logId =
-        InternalLogId.allocate(getClass(), address != null ? address.toString() : "No address");
+    this.logId = InternalLogId.allocate(getClass(), addresses.isEmpty() ? "No address" :
+        String.valueOf(addresses));
+    this.bossExecutor = bossGroup.next();
   }
 
   @Override
   public SocketAddress getListenSocketAddress() {
-    if (channel == null) {
+    Iterator<Channel> it = channelGroup.iterator();
+    if (it.hasNext()) {
+      return it.next().localAddress();
+    } else {
       // server is not listening/bound yet, just return the original port.
-      return address;
+      return addresses.isEmpty() ? null : addresses.get(0);
     }
-    return channel.localAddress();
+  }
+
+  @Override
+  public List<SocketAddress> getListenSocketAddresses() {
+    List<SocketAddress> listenSocketAddresses = new ArrayList<>();
+    for (Channel c: channelGroup) {
+      listenSocketAddresses.add(c.localAddress());
+    }
+    return listenSocketAddresses;
   }
 
   @Override
   public InternalInstrumented<SocketStats> getListenSocketStats() {
-    return listenSocketStats;
+    return listenSocketStatsList.isEmpty() ? null : listenSocketStatsList.get(0);
+  }
+
+  @Override
+  public List<InternalInstrumented<SocketStats>> getListenSocketStatsList() {
+    return listenSocketStatsList;
   }
 
   @Override
   public void start(ServerListener serverListener) throws IOException {
     listener = checkNotNull(serverListener, "serverListener");
 
-    ServerBootstrap b = new ServerBootstrap();
+    final ServerBootstrap b = new ServerBootstrap();
     b.option(ALLOCATOR, Utils.getByteBufAllocator(forceHeapBuffer));
     b.childOption(ALLOCATOR, Utils.getByteBufAllocator(forceHeapBuffer));
-    b.group(bossGroup, workerGroup);
+    b.group(bossExecutor, workerGroup);
```

Comment:
What's the reason for this change?

Reply:
@ejona86  suggested before(https://github.com/grpc/grpc-java/issues/6641#issuecomment-734397031), and I thought it is a brilliant idea in terms of thread safety:
> If NettyServer is handling the multiple ports, then we can do all the binds as one "atomic" operation on the event loop. This depends on a behavior of Channel that it executes the operation immediately if already on the event loop.